HLTrigger/special/test/testLogMonitorFilter.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Pass in name and status
+function die { echo $1: status $2 ; echo === Log file === ; cat ${3:-/dev/null} ; echo === End log file === ; exit $2; }
+
+
+TESTDIR=${LOCALTOP}/src/HLTrigger/special/test
+
+cmsRun ${TESTDIR}/testLogMonitorFilter.py >& log_log_monitor_filter || die "Failure using testLogMonitorFilter.py" $? log_log_monitor_filter
+
```

Comment:
Do you mean it fails when run on the old code (which I expect) or is it failing for the new code (which is not supposed to happen)?

Reply:
Both, in my checks. But yes, the comment was about the new implementation.

Reply:
From the PR test, it shows the new unit test does work with the new code. The unit test will fail with the old code because I slightly changed the meaning of `Prescale` in the summary. Before, it meant what `Prescale` would be used if another event would be processed by the job. I changed it to mean the `Prescale` used on the very last processed event. Since the number of events processed by the test job was an exact multiple of the `Prescale` value, the last run event had a `Prescale` of 100 applied to it but the next event would have had 1000.

Reply:
The report from the PR says the unit test passes. From https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-704425/21510/unitTests/log.txt

```
===== Test "testHLTriggerspecialLogMonitorFilter" ====

---> test testHLTriggerspecialLogMonitorFilter succeeded
TestTime:8
^^^^ End Test testHLTriggerspecialLogMonitorFilter ^^^^
>> Tests for package HLTrigger/special ran.
```

Reply:
I agree, the unit test works as is; I was not questioning that (sorry, I should have been clearer). To expand on my original comment: what I did was to check out this PR locally and to run the unit test (which works), and after that I manually edited `testLogMonitorFilter.py` to use `numberOfThreads = 2` (or higher); only then I saw the unit test fail. I guess my question is whether one should expect the unit test to pass even when `numberOfThreads > 1`.

Reply:
I'm afraid not. The test has two paths defined
```
process.path        = cms.Path(process.warning + process.error + process.combined + process.reject)
process.logmonitor  = cms.Path(process.hltLogMonitorFilter)
```

As the modules `warning`, `error` and `combined` issue Log messages but are `EDAnalyzers`, there is no explicit data dependencies between those modules and `hltLogMonitorFilter` so the framework is allowed to run them in what ever order it finds useful. With more than one thread, that could be any arbirary ordering.

The test could be changed to add another EDFilter to beginning of the `process.logmonitor` path where that new EDFilter would depend on the results of the `process.path` Path. In that way, all modules on `process.path` would have to finish running before `process.logmonitor` is executed.

Reply:
I see. Thanks again for clarifying.

Reply:
I appreciate that you took the effort to checkout the code and try it!